fedimint-core/src/config.rs

```diff
@@ -239,20 +239,23 @@ impl ClientConfig {
     ) -> anyhow::Result<ClientConfig> {
         debug!("Downloading client config from {:?}", invite_code);
 
-        for _ in 0..10 {
+        let mut attempt = 1;
+        loop {
             match Self::try_download_client_config(invite_code).await {
                 Ok(cfg) => {
                     return Ok(cfg);
                 }
                 Err(error) => {
-                    debug!("Failed to download client config {:?}", error);
+                    debug!(?error, %attempt, "Failed to download client config");
+                    if attempt == 10 {
+                        return Err(error.context("Failed to download client config"));
+                    }
 
                     sleep(Duration::from_millis(500)).await;
                 }
             }
+            attempt += 1;
         }
-
-        Err(anyhow!("Failed to download client config"))
     }
```

Comment:
Can we make this a utility function (or do we have one already)?

```rust
// Return Ok on first success or last Err after a given number of retries
async fn(usize, Fn() -> Future<Output=Result<T, E>>) -> Result<T, E>
```

That would at least encapsulate that mutability (`attempt`) in the loop a bit.

Reply:
fixed